src/ros.cpp

```diff
@@ -98,9 +98,12 @@ void RosPosePublisher::setFrame(QString frame)
 void RosPosePublisher::publish(){
     double x,y, theta;
     if (_origin) {
-        x = _origin->mapToItem(_target,QPoint(0,0)).x() * _pixel2meter;
-        y = _origin->mapToItem(_target,QPoint(0,0)).y() * _pixel2meter;
+        x = _origin->mapFromItem(_target,QPoint(0,0)).x() * _pixel2meter;
+        y = - _origin->mapFromItem(_target,QPoint(0,0)).y() * _pixel2meter;
         theta = (_target->rotation() - _origin->rotation()) * M_PI/180;
+        cout <<"origin "<<_origin->x() * _pixel2meter<<" "<<_origin->y() * _pixel2meter<<endl;
+        cout <<"target "<<_target->x() * _pixel2meter<<" "<<_target->y() * _pixel2meter<<endl;
+        cout <<"mapped "<<x<< " " << y<<endl;
```

Comment:
Please no `cout`... use ROS logging if output is really needed (but I believe it is not in this case)